File: src/finding.py
from gql.client import AsyncClientSession

from .gql import gql
from .analyze import Finding


async def create_finding(client: AsyncClientSession, finding: Finding):
    query = gql(
        """
        mutation CreateFinding($requestId: ID!, $input: CreateFindingInput!) {
            # CODE
        }
        """
    )
    description = "Found reflected parameters in reponse:\n"
    for key, value in finding.parameters:
        description += f"- {key}: {value}\n"

    finding = await client.execute(
        query,
        variable_values={
            "requestId": finding.id,
            "input": {
                # CODE
            },
        },
    )

    if finding["createFinding"]["finding"]:
        print(f"[-] Created finding {finding['createFinding']['finding']['id']}")

File: packages/backend/src/finding.ts
import type { SDK } from "caido:plugin";

import type { Finding } from "./analyse";

export async function createFinding(sdk: SDK, finding: Finding) {
  let description = "Found reflected parameters in reponse:\n";
  for (let [key, value] of finding.parameters) {
    description += `- ${key}: ${value}\n`;
  }

  const result = {}; // CODE

  // @ts-ignore
  sdk.console.log(`Finding created with ID ${result.getId()}`);
}
